clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/tests/GvcfTests.scala

```diff
@@ -375,4 +375,172 @@ trait GvcfTests { self: BaseIntegrationSpec =>
       }
     }
   }
+
+  it should "move gvcfs in GCP" in {
+    val project = s"abcdefg$randomId"
+    val sample = s"sample$randomId"
+    val version = 3
+
+    val fileContents = s"$randomId --- I am a dummy gvcf --- $randomId"
+    val cloudPath = rootTestStorageDir.resolve(
+      s"gvcf/$project/$sample/v$version/$randomId.vcf.gz"
+    )
+    val cloudPath2 = cloudPath.getParent.resolve(s"moved/$randomId.vcf/gz")
+
+    val key = TransferGvcfV1Key(Location.GCP, project, sample, version)
+    val metadata =
+      TransferGvcfV1Metadata(gvcfPath = Some(cloudPath.toUri.toString))
+
+    // Clio needs the metadata to be added before it can be moved.
+    val _ = Files.write(cloudPath, fileContents.getBytes)
+    val result = for {
+      _ <- runUpsertGvcf(key, metadata)
+      _ <- runClient(
+        ClioCommand.moveGvcfName,
+        "--location",
+        Location.GCP.entryName,
+        "--project",
+        project,
+        "--sample-alias",
+        sample,
+        "--version",
+        version.toString,
+        "--destination",
+        cloudPath2.toUri.toString
+      )
+    } yield {
+      Files.exists(cloudPath) should be(false)
+      Files.exists(cloudPath2) should be(true)
+      new String(Files.readAllBytes(cloudPath2)) should be(fileContents)
+    }
+
+    result.andThen[Unit] {
```

Comment:
This should be run even if the test fails or throws, in Java I'd use a `finally`

Reply:
Yup, `andThen` is the `finally` of the `Future` API.

Reply:
OK. I thought the `yield` body could be evaluated before `andThen()` is called. But I guess since it throws, it would have to be evaluated in the current thread to work properly.